DcShare/Helper/ShareAttachment.swift

```diff
@@ -63,6 +66,37 @@ class ShareAttachment {
         }
     }
 
+    // for now we only support GIF
+    func createAnimatedImageMsg(_ item: NSItemProvider) {
+        item.loadItem(forTypeIdentifier: kUTTypeGIF as String, options: nil) { data, error in
+            var result: SDAnimatedImage?
+            switch data {
+            case let animatedImageData as Data:
+                result = SDAnimatedImage(data: animatedImageData)
+            case let url as URL:
+                result = SDAnimatedImage(contentsOfFile: url.path)
+            default:
+                self.dcContext.logger?.debug("Unexpected data: \(type(of: data))")
+            }
+            if let result = result, let animatedImageData = result.animatedImageData {
+                let path = DcUtils.saveAnimatedImage(data: animatedImageData, suffix: "gif")
+                let msg = DcMsg(viewType: DC_MSG_GIF)
+                msg.setFile(filepath: path, mimeType: "image/gif")
+                let pixelSize = result.imageSizeInPixel()
+                msg.setDimension(width: pixelSize.width, height: pixelSize.height)
+                self.messages.append(msg)
+                self.delegate?.onAttachmentChanged()
+                if self.imageThumbnail == nil {
+                    self.imageThumbnail = result.scaleDownImage(toMax: self.thumbnailSize)
+                    self.delegate?.onThumbnailChanged()
+                }
+                if error != nil {
+                    self.dcContext.logger?.error(error?.localizedDescription ?? "Could not load share item as image")
```

Comment:
minor thing, but the `"Could not load share item as image"` is unreachable.
Mabye better:
`if let error = error`, so we don't need the optional chaining.